src/functions/durableOrchestration.ts

```diff
@@ -0,0 +1,25 @@
+import { HttpRequest, InvocationContext } from '@azure/functions';
+import * as df from 'durable-functions';
+import { DurableOrchestrationClient, IOrchestrationFunctionContext } from 'durable-functions/lib/src/classes';
+
+df.orchestration('DurableFunctionsOrchestratorJS', function* (context: IOrchestrationFunctionContext) {
+    const outputs = [];
+    outputs.push(yield context.df.callActivity('Hello', 'Tokyo'));
+    outputs.push(yield context.df.callActivity('Hello', 'Seattle'));
+    outputs.push(yield context.df.callActivity('Hello', 'Cairo'));
+
+    return outputs;
+});
+
+df.activity('Hello', (context: InvocationContext, name: string) => {
+    return `Hello, ${name}`;
+});
+
+df.httpClient(
+    'DurableFunctionsHttpStart',
+    async (context: InvocationContext, req: HttpRequest, client: DurableOrchestrationClient) => {
+        const instanceId = await client.startNew(req.query.get('functionName'), undefined, req.body);
```

Comment:
In the old templates, the route was set to `orchestrators/{functionName}` and we'd use `req.params.functionName` here. I don't see any particular reason the query is better, so I'd prefer to keep it as a param

Similar feedback for the entity client

Reply:
The reason is because `df.httpClient()` (at least the way I currently designed it) sets a bunch of defaults to just streamline the code, so one consequence is that it doesn't allow setting a custom route, so now the way to get the function name would be through the query not the params

Reply:
Of course, one could still use the more generic `df.client()` to set the custom route and do things the old way. I can change one of the client functions to do that to demonstrate

Reply:
> The reason is because df.httpClient() (at least the way I currently designed it) sets a bunch of defaults to just streamline the code, so one consequence is that it doesn't allow setting a custom route

We streamlined the http triggers, but that was because they
1. Are by far the most popular 
2. Are used heavily by brand new users
3. Are supported by many _other_ frameworks (express, etc.) and we wanted to be idiomatic

No other trigger was streamlined in that way and I don't think durable should be either. Sure we can have a few smart defaults, but I think a user should always pass in an options object at minimum (And I think they'll be fine with that).

```
df.client('durableEntityStart1', trigger.http({ route: 'entity/{id}' }), output.http({}), clientHandler);
```

This does not match the pattern used by all other triggers. Durable should be consistent and use an options object

Reply:
> We streamlined the http triggers, but that was because they
> 
> 1. Are by far the most popular
> 2. Are used heavily by brand new users
> 3. Are supported by many _other_ frameworks (express, etc.) and we wanted to be idiomatic

Those same arguments could be made here too. Technically, what I'm streamlining here _is_ the HTTP trigger, just for client functions. 
 
> I think a user should always pass in an options object at minimum (And I think they'll be fine with that).

Sure, that is accomplished by the "complex" variant of the triggers. I agree the names may not be the most intuitive or make the most sense. I've added another file example to showcase that though for activities and clients.


> ```
> df.client('durableEntityStart1', trigger.http({ route: 'entity/{id}' }), output.http({}), clientHandler);
> ```
> 
> This does not match the pattern used by all other triggers. Durable should be consistent and use an options object

I can be in favor of dropping `df.client()` for `df.clientComplex()`, renaming it to `df.client()` ofc.